### PR TITLE
don't handle meta files in details

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Remove the display and interaction with .meta
files from details entries.

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

## Type of change

- [x] :bug: Bugfix

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/internal/connector/onedrive/collection.go
Comment (meain) [inline src/internal/connector/onedrive/collection.go:418]: The filenames are pretty stable, they will always have one of `.data`, `.meta` or `.dirmeta` suffixes. Should we just make use of that instead of adding new filed in the driveInfo? This might be useful long term though.

File: src/pkg/backup/details/details.go
Comment (meain) [inline src/pkg/backup/details/details.go:113]: nit: `SliceMetaFiles` seems like an odd name. How about `GetDataFiles` or `FilterDataFiles`.

File: src/pkg/backup/details/details.go
Comment (meain) [inline src/pkg/backup/details/details.go:128]: We need to do this based on the backup version. For backup version 0, there is never any meta files. All files, even the ones that end with `.meta` are data files and for all other versions, if the suffix is `.meta` or `.dirmeta`, they are metadata files.

File: src/pkg/backup/details/details_test.go
Comment (meain) [inline src/pkg/backup/details/details_test.go:232]: It might not affect this test, but we should always have a suffix of `.data`, `.meta` or `.dirmeta` for the locationRef for any newer backups.

**Review by meain — CHANGES_REQUESTED** [review]

File: src/internal/connector/onedrive/collection.go
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/collection.go:418]: Stability isn't the concern.  Cross-contamination is my worry.  If we back-up a file named `foo.meta` that has nothing to do with our meta scheme, it could cause problems.  Sure, right now it'll be called `foo.meta.data`, but if we end up changing that naming scheme in the future, this might still catch us by surprise.

And/or we can drop it when we move to storing files by ID.

File: src/pkg/backup/details/details_test.go
Comment (ryanfkeepers) [reply src/pkg/backup/details/details_test.go:232]: Let's not cross wires, and come back to that in a different PR.

File: src/pkg/backup/details/details.go
Comment (meain) [reply src/pkg/backup/details/details.go:128]: My concern is that if we have files from older backup with a suffix of `.meta` in the original file name, it can end up getting treated as a metadata file even though it is not.

File: src/pkg/backup/details/details_test.go
Comment (meain) [reply src/pkg/backup/details/details_test.go:232]: What I meant by newer backups is that any backups done with the latest version of the code.

File: CHANGELOG.md
Comment (meain) [inline CHANGELOG.md:14]:
```suggestion
- .meta files no longer appear in the backup details.
```

File: src/pkg/backup/details/details.go
Comment (meain) [inline src/pkg/backup/details/details.go:129]: We should also check for dirmeta.

File: src/pkg/backup/details/details_test.go
Comment (ryanfkeepers) [reply src/pkg/backup/details/details_test.go:232]: Understood.  What I mean is that this test can be brought up to date in a different PR.

Also, now that I'm re-reading your comment, `locationRef` should _not_ contain those values.  It's entirely a folder reference, and contains no file data.  It should be unaware of both data and meta categorization.

File: src/pkg/backup/details/details.go
Comment (ryanfkeepers) [reply src/pkg/backup/details/details.go:128]: Agreed, thus the todo.  I can add a known issue to the changelog, if that helps.  Also, not saying v2 needs to be a long follow.  You should probably tail this PR with the change immediately.  But I think this is safe enough for the immediate case, and it's too late for me to push that next part.

File: src/pkg/backup/details/details.go
Comment (meain) [inline src/pkg/backup/details/details.go:39]: We should do this filter in OneDrive before we sent the `DetailsModel` to `PrintEntries` as it will otherwise affect others like Exchange which does not even have the idea of `.meta` files.

**Comment by meain** [comment]
I have a follow up to this https://github.com/alcionai/corso/pull/2610 which fixes how we mark an item as a metadata item.

File: src/pkg/repository/repository.go
Comment (meain) [inline src/pkg/repository/repository.go:367]: Current version in prod is version 2, the change being it has name in meta. You should be able to see this if you check the version number of the current backups. This is why I had > and < checks.

Comment (meain) [reply src/pkg/repository/repository.go:367]: Also, if you see `Restore2IsMetaMarker` is current set to 3. So we need both 1 and 2 to run this.

File: src/internal/version/restore.go
Comment (meain) [inline src/internal/version/restore.go:13]:
```suggestion
	Restore3IsMetaMarker = 3
```

`RestoreCurrNameInMeta` is technically 2, but holding off for now.

**Comment by meain** [comment]
@ashmrtn If you do have time, do take a look at the versioning related changes here.

**Review by meain — APPROVED** [review]
Seems to be good now. I would suggest waiting for Ash to take a look at it as well. There is a typecheck issue in link related to the file moves. You might wanna take a look at that as well. Approving it to unblock.

File: src/internal/version/restore.go
Comment (ashmrtn) [inline src/internal/version/restore.go:5]: I like having the comments on these. A few small non-blocking nits though:
* comments don't note that they actually apply to OneDrive stuff
* splitting restore and backup versions is a bit confusing. They're actually related to each other but having them different like this makes me think they refer to different things

**Review by ashmrtn — APPROVED** [review]

File: src/internal/version/restore.go
Comment (ryanfkeepers) [reply src/internal/version/restore.go:5]: > comments don't note that they actually apply to OneDrive stuff

Added

> splitting restore and backup versions is a bit confusing. They're actually related to each other but having them different like this makes me think they refer to different things

They were already split in the code, this is just migration.  We can make a tech-debt follow-up to unify them if needed.

Comment (ryanfkeepers) [reply src/internal/version/restore.go:5]: Ah, as soon as I looked at the code I grokked what you were saying.  Sorry, my brain is all a scatter today.  I've made a pass at the locations and naming scheme for the labels.  Should hopefully fit what you're looking for.

File: src/internal/connector/graph_connector_onedrive_test.go
Comment (ryanfkeepers) [inline src/internal/connector/graph_connector_onedrive_test.go:235]: This took me waaaaaaaaaaaaaay too long to find.

**Pull request merged**
